app/controllers/users_controller.rb

```diff
@@ -146,6 +146,29 @@ def username
     end
   end
 
+  def toggle_second_factor
+    params.require(:token)
+    user = fetch_user_from_params
+    unless user.authenticate_otp(params[:token])
+      RateLimiter.new(nil, "second-factor-min-#{request.remote_ip}", 3, 1.minute).performed!
+      render json: { error: I18n.t("login.invalid_second_factor_code") }
+      return
+    end
+    if user.totp_enabled?
+      user.user_second_factor.delete
+      Jobs.enqueue(
+        :critical_user_email,
+        type: :account_second_factor_disabled,
+        user_id: user.id
+      )
+    else
+      user.create_totp
+      user.user_second_factor.enabled = true
+      user.user_second_factor.save!
+    end
+    render json: {}
```

Comment:
this either worked or not ... it should probably send through json_success like other spots do